### PR TITLE
Working response_handler.py

### PR DESCRIPTION
**Pull request by littleMissMae**
@sewagodimo, could you please review my PR :)

### CODE REVIEW

File: app/gsheets_reader/gsheets_reader.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return value
+
+def get_definition(term):
+    ''' Access the google sheet and return the
+        key value for the term
+        
+        term - term to be defined
+        key - definition to return
+        sheet - name of google sheet to read'''
+    value = None
```
Comment (sewagodimo) [inline app/gsheets_reader/gsheets_reader.py]: don't name vars value, rather name what that value will mean or be used for

File: app/gsheets_reader/gsheets_reader.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    df = None
```
Comment (sewagodimo) [inline app/gsheets_reader/gsheets_reader.py]: even df is not very helpful. It's the value get_gsheet so name is something along those lines

File: app/gsheets_reader/gsheets_reader.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    try:
+        df = get_gsheet()    
+    except:
+        print("Sheet could not be fetched")
```
Comment (sewagodimo) [inline app/gsheets_reader/gsheets_reader.py]: why not just through an actual exeption, a print is not really a good idea, especially in production. It's good for testing locally

File: app/gsheets_reader/gsheets_reader.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        print("*** gsheet fetched")
```
Comment (sewagodimo) [inline app/gsheets_reader/gsheets_reader.py]: also in the same exception you say that you did not find it then you found it

File: app/gsheets_reader/gsheets_reader.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        
+
+    try:
+        value = get_cell_data(term, 'Definition', df)
+    except Exception as e: 
+        print("Value could not be extracted")
```
Comment (sewagodimo) [inline app/gsheets_reader/gsheets_reader.py]: this one two

File: app/gsheets_reader/gsheets_reader.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+# 1) Go to GCP > APIs & Services > Credentials >
```
Comment (sewagodimo) [inline app/gsheets_reader/gsheets_reader.py]: this could be in a readme

File: app/gsheets_reader/quickstart.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+if not creds or creds.invalid:
+    flow = client.flow_from_clientsecrets('client_secret.json', SCOPES)
+    creds = tools.run_flow(flow, store)
+service = build('sheets', 'v4', http=creds.authorize(Http()))
+
+# Call the Sheets API
+SPREADSHEET_ID = '1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms'
+RANGE_NAME = 'Class Data!A2:E'
+result = service.spreadsheets().values().get(spreadsheetId=SPREADSHEET_ID,
+                                             range=RANGE_NAME).execute()
```
Comment (sewagodimo) [inline app/gsheets_reader/quickstart.py:21]: accidental extra line

File: flo_detect_intent.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import uuid
+
+import dialogflow
+
+# [MY IMPORTS]
+import os
+#os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = './LittleSis-3089dc6d6fb8.json'
+os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = './v2_sa.json'
+
+# [END import_libraries]
```
Comment (sewagodimo) [inline flo_detect_intent.py]: this isnt needed

File: flo_detect_intent.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+
+# [START dialogflow_detect_intent_text]
```
Comment (sewagodimo) [inline flo_detect_intent.py]: these comments won't be needed but they are super cute

File: flo_detect_intent.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+def detect_intent_texts(project_id, session_id, texts, language_code):
+    """Returns the result of detect intent with texts as inputs.
+
+    Using the same `session_id` between requests allows continuation
+    of the conversaion."""
+    session_client = dialogflow.SessionsClient()
```
Comment (sewagodimo) [inline flo_detect_intent.py]: add a space after the docstring just to make it look neat

File: flo_detect_intent.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    for text in texts:
+        text_input = dialogflow.types.TextInput(
+            text=text, language_code=language_code)
+
+        query_input = dialogflow.types.QueryInput(text=text_input)
+
+        response = session_client.detect_intent(
+            session=session, query_input=query_input)
+
+        print('=' * 20)
```
Comment (sewagodimo) [inline flo_detect_intent.py]: are these print statements important for the application? Else you can just remove them before committing or leave them out of the commit

File: flo_detect_intent.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        print('Query text: {}'.format(response.query_result.query_text))
+        print('Detected intent: {} (confidence: {})\n'.format(
+            response.query_result.intent.display_name,
+            response.query_result.intent_detection_confidence))
+        print('Fulfillment text: {}\n'.format(
+            response.query_result.fulfillment_text))
+
+        return response
+    
+# [END dialogflow_detect_intent_text]
```
Comment (sewagodimo) [inline flo_detect_intent.py]: you can remov e this too

File: response_handler.py [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import uuid
+from flask import Flask, request, make_response, jsonify
+from flo_detect_intent import detect_intent_texts
+import rp_json
+
+app = Flask(__name__)
+
+def get_df_response(message):
+  #  agent_id = 'littlesis-a5948'
```
Comment (sewagodimo) [inline response_handler.py]: can also be removed

Comment (sewagodimo) [reply response_handler.py]: or just add a doctstring

**Review by sewagodimo — CHANGES_REQUESTED** [review]
So just try to keep the code base clean and it will be easier to review it

File: app/gsheets_reader/quickstart.py (hunk 7 above)
Comment (littleMissMae) [reply app/gsheets_reader/quickstart.py:21]: @sewagodimo , isn't this just to accommodate for the line wrap of `range=RANGE_NAME...`
?

File: flo_detect_intent.py (hunk 11 above)
Comment (littleMissMae) [reply flo_detect_intent.py]: I believe these are just part of the Dialogflow python code snippet. They might be helpful for future use -- should I just comment them out?

**Pull request merged**